deployment/terraform/assets/cluster.tf

```diff
@@ -77,6 +82,63 @@ resource "aws_instance" "app_server" {
   }
 }
 
+
+data "aws_iam_policy_document" "metrics_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "metrics_role" {
+  name               = "${var.cluster_name}-metrics-role"
+  assume_role_policy = data.aws_iam_policy_document.metrics_assume_role.json
+}
+
+resource "aws_iam_instance_profile" "metrics_profile" {
+  name = "metrics_profile"
+  role = aws_iam_role.metrics_role.name
+}
```

Comment:
@agarciamontoro Heads up this seems to be causing issues on my side, apparently conflicting with other deployments. Getting the following error when deploying:

```
info  [2024-09-25 15:49:43.890 -06:00] │ Error: creating IAM Instance Profile (metrics_profile): operation error IAM: CreateInstanceProfile, https response error StatusCode: 409, RequestID: 5a5d50ea-fa2d-411b-b185-2009a4975393, EntityAlreadyExists: Instance Profile metrics_profile already exists. caller="terraform/engine.go:97"
```

Reply:
Of course! :facepalm: Thank you for catching this!